src/app/services/account.service.ts

```diff
@@ -0,0 +1,61 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { BehaviorSubject, first,Subject,tap } from 'rxjs';
+import { User } from '../Users';
+import { Router } from '@angular/router';
+
+@Injectable({
+  providedIn: 'root'
+})
+ export class AccountService {
+  public redirectUrl?: string; 
+  private currentUserSubject = new BehaviorSubject<User | string>("default"); 
+  private localStorageUserSubject = new BehaviorSubject<User | string | null>("default");
+  constructor(private http: HttpClient, private router: Router) {  
+  }
+
+  getCurrentUserSubject(): BehaviorSubject<User | string> {
+    return this.currentUserSubject; 
+  }
+  getlocalStorageUserSubject(): BehaviorSubject<any> {
+   return this.localStorageUserSubject; 
+  }
+
+  login(email: string, password: string){
+    return this.http.post<User>('api/authenticate', {email, password}).pipe(
+      first(), //emitt 1 item and close the Observable 
+      tap(user =>{
+       localStorage.setItem('currentUser', JSON.stringify(user));
+       this.updateLocalStorageSubject(); 
+       return user; 
+      })
+    ); 
+  }
+
+  logOut():void{
+    console.log("logOut() called")
+    localStorage.removeItem('currentUser');
+    //this.currentUserSubject.next("default");
+    this.localStorageUserSubject.next('default');
+    this.router.navigate(['/login']);
+  }
+
+  signUp(userData:User){
+  return this.http.post<User>('api/register', userData ) 
+  }
+
+  isLoggedIn():boolean{
+    //return false if null or undefined, true otherwise!
+    console.log("local user",localStorage.getItem('currentUser') )
+     return !!localStorage.getItem('currentUser');  
+  }
```

Comment:
Consistent spacing is important. Make it 2 or 4 but continue to do it through the app. remove whitespace between parentheses. To help with this you should make sure you have a linter installed. enter `ng lint` in your terminal to make sure it is set up. Go through the prompts to install it if it is not. It will help you with these spacing inconsistencies.

Reply:
I implemented a bunch of layout rules using eslint. I did not know what lint was, its extremely useful. Code should look a bit cleaner now, thanks.